paimon-core/src/main/java/org/apache/paimon/table/source/snapshot/StartingScanner.java

```diff
@@ -22,38 +22,74 @@
 import org.apache.paimon.table.source.TableScan;
 import org.apache.paimon.utils.SnapshotManager;
 
-import javax.annotation.Nullable;
-
 import java.util.Collections;
 import java.util.List;
 
 /** Helper class for the first planning of {@link TableScan}. */
 public interface StartingScanner {
 
-    @Nullable
     Result scan(SnapshotManager snapshotManager, SnapshotSplitReader snapshotSplitReader);
 
     /** Scan result of {@link #scan}. */
-    class Result {
+    interface Result {
+        long snapshotId();
+
+        List<DataSplit> splits();
+    }
+
+    /** A null Result. */
+    class NullResult implements Result {
```

Comment:
`NoSnapshot`, and add comment: Currently, there is no snapshot, need to wait for the snapshot to be generated.